src/Compilers/VisualBasic/Portable/Symbols/NamedTypeSymbol.vb

```diff
@@ -448,6 +449,108 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             End Get
         End Property
 
+        Public Overrides Function GetHashCode() As Integer
+            If Me.SpecialType = SpecialType.System_Object Then
+                Return SpecialType.System_Object
+            End If
+
+            Return RuntimeHelpers.GetHashCode(OriginalDefinition)
+        End Function
+
+        ''' <summary>
+        ''' Compares this type to another type.
+        ''' </summary>
+        Friend Overrides Function Equals(t2 As TypeSymbol, comparison As TypeCompareKind) As Boolean
+            Debug.Assert(Not Me.IsTupleType)
+            If DirectCast(t2, Object) Is Me Then
+                Return True
+            End If
+
+            If DirectCast(t2, Object) Is Nothing Then
+                Return False
+            End If
+
+            If (comparison And TypeCompareKind.IgnoreDynamic) <> 0 Then
+                If t2.TypeKind = TypeKind.Dynamic Then
+                    If Me.SpecialType = SpecialType.System_Object Then
+                        Return True
+                    End If
+                End If
+            End If
```

Comment:
Thank you! Fixed.

---
In reply to: [117133293](https://github.com/dotnet/roslyn/pull/19283#discussion_r117133293) [](ancestors = 117133293)